pkg/controllers/sync/placement.go

```diff
@@ -22,94 +21,24 @@ are Copyright 2023 The KubeAdmiral Authors.
 package sync
 
 import (
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/util/sets"
 
 	fedcorev1a1 "github.com/kubewharf/kubeadmiral/pkg/apis/core/v1alpha1"
-	"github.com/kubewharf/kubeadmiral/pkg/controllers/util"
 )
 
-// computeNamespacedPlacement determines placement for namespaced
-// federated resources (e.g. FederatedConfigMap).
-//
-// If KubeFed is deployed cluster-wide, placement is the intersection
-// of the placement for the federated resource and the placement of
-// the federated namespace containing the resource.
-//
-// If KubeFed is limited to a single namespace, placement is
-// determined as the intersection of resource and namespace placement
-// if namespace placement exists.  If namespace placement does not
-// exist, resource placement will be used verbatim.  This is possible
-// because the single namespace by definition must exist on member
-// clusters, so namespace placement becomes a mechanism for limiting
-// rather than allowing propagation.
-func computeNamespacedPlacement(
-	resource, namespace *unstructured.Unstructured,
-	clusters []*fedcorev1a1.FederatedCluster,
-	limitedScope bool,
-) (selectedClusters sets.String, err error) {
-	resourceClusters, err := computePlacement(resource, clusters)
-	if err != nil {
-		return nil, err
-	}
-
-	if namespace == nil {
-		if limitedScope {
-			// Use the resource placement verbatim if no federated
-			// namespace is present and KubeFed is targeting a
-			// single namespace.
-			return resourceClusters, nil
-		}
-		// Resource should not exist in any member clusters.
-		return sets.String{}, nil
-	}
-
-	namespaceClusters, err := computePlacement(namespace, clusters)
-	if err != nil {
-		return nil, err
-	}
-
-	// If both namespace and resource placement exist, the desired
-	// list of clusters is their intersection.
-	return resourceClusters.Intersection(namespaceClusters), nil
-}
-
 // computePlacement determines the selected clusters for a federated
 // resource.
 func computePlacement(
```

Comment:
Would it be better to move these functions to `resource.go`? They are only used by `FederatedResource`.

Reply:
What's the issue with them being in a separate file? I think `resource.go` is pretty cluttered.

Reply:
This is a NIT, I just thought they were 2 relatively small and simple functions that look out-of-place in their own file.